addons/sale_expense/views/hr_expense_views.xml

```diff
@@ -6,7 +6,7 @@
         <field name="inherit_id" ref="hr_expense.hr_expense_view_form"/>
         <field name="arch" type="xml">
             <xpath expr="//field[@name='employee_id']" position="after">
-                <field name="sale_order_id" context="{'sale_show_partner_name': True}"/>
+                <field name="sale_order_id" context="{'sale_show_partner_name': True}" groups="sales_team.group_sale_salesman"/>
```

Comment:
@jem-odoo I'm seeing this will raise again the error, as the field tries to be loaded any way. You should put the group at view level, not at field level.